build.sh

```diff
@@ -1,16 +1,30 @@
 #!/bin/bash
 set -xe
+. has_to_execute_integration.sh
 
 # You can run it from any directory.
 DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
 PROJECT_DIR=$DIR/
 
 # This will: compile the project, run lint, run tests under JVM, package apk, check the code quality and run tests on the device/emulator.
-"$PROJECT_DIR"/gradlew --no-daemon clean
-"$PROJECT_DIR"/gradlew --no-daemon build -Dscan
-"$PROJECT_DIR"/gradlew --no-daemon test
-"$PROJECT_DIR"/gradlew --no-daemon connectedAndroidTest -Pandroid.testInstrumentationRunnerArguments.size=small
-if [ "$TRAVIS_BRANCH" == "development" ] || [ "$TRAVIS_BRANCH" == "master" ];
-then "$PROJECT_DIR"/gradlew --no-daemon connectedAndroidTest -Pandroid.testInstrumentationRunnerArguments.size=medium; fi
-if [ "$TRAVIS_BRANCH" == "master" ];
-then "$PROJECT_DIR"/gradlew --no-daemon connectedAndroidTest -Pandroid.testInstrumentationRunnerArguments.size=large; fi
\ No newline at end of file
+if [ "$TEST_SUITE" == "units" ];
+then
+"$PROJECT_DIR"/gradlew clean
+"$PROJECT_DIR"/gradlew build -Dscan; fi
+
+if [ "$EXECUTE_INTEGRATION" == "true" ]
+then
+android-wait-for-emulator
+sleep 180
+adb devices
+adb shell input keyevent 82 &
+    if [ "$TEST_SUITE" == "integration" ]
+    then
+    "$PROJECT_DIR"/gradlew connectedAndroidTest -Pandroid.testInstrumentationRunnerArguments.size=small
+    fi
+    if [ "$TEST_SUITE" == "integration_large" ]
+    then
+    "$PROJECT_DIR"/gradlew connectedAndroidTest -Pandroid.testInstrumentationRunnerArguments.size=medium
```

Comment:
The command ./build.sh integration_large exited with 13, I think we should research this 137 error code, please look [this](https://github.com/travis-ci/travis-ci/issues/4192)